py/farm_ng/core/events_file_reader.py

```diff
@@ -95,19 +107,28 @@ def close(self) -> bool:
         return self.is_closed()
 
     def _reset_event_index(self) -> None:
-        self._event_index = defaultdict(list)
+        self._event_index = []
 
-    def uris(self) -> List[Uri]:
-        if not self._event_index:
+    def get_uris(self) -> List[Uri]:
+        if len(self._event_index) == 0:
             self._build_event_index()
-        return [string_to_uri(key) for key in sorted(self._event_index.keys())]
+
+        uris_set: Set[EventLogPosition] = set()
+        for event_log in self.event_index:
+            uris_set.add(uri_to_string(event_log.event.uri))
+
+        return [string_to_uri(x) for x in sorted(uris_set)]
```

Comment:
Can you not make a set of uri's to begin with?

Reply:
just replicating the previous behavior where we based on a dict with unique uris, in the this new approach using EventLogPosition we build the index a list not dict